wordpress-mobile/gutenberg-mobile
# Improving accessibility on Image Block (deselected)

Part of #476 

This PR improves accessibility (screen reader) for the Image Block. In particular when it's deselected.
`gutenberg` side PR: https://github.com/WordPress/gutenberg/pull/14713

The current behavior is:
Empty image block:
- **Label:** "Image block. Empty."
- **Hint:** "Double tap to select an image."

Image block with image:
- **Label:** "Image block. {alt}."
- **Hint:** ""Double tap to edit the image."

cc @rachelmcr 

When the block is not selected, it's captured by the screen reader as one whole element, but when it is selected, it opens the inner controls, including the inner tool bar.

### To test (iOS):
- Activate VoiceOver: https://github.com/wordpress-mobile/WordPress-iOS/wiki/Using-VoiceOver

### To test (Android):
- Activate TalkBack: https://www.wikihow.com/Enable-TalkBack-on-Your-Android
  - In my case, the system had a handy tutorial on how to use it.
  - If your version doesn't have it, it's actually very similar to [VoiceOver.](https://www.wikihow.com/Enable-TalkBack-on-Your-Android)

### Common test steps:
- Select an empty Image Block.
- [ ] Check that it reads as described before, including label and hint. (Hint takes a couple of seconds on iOS).
- Activate the selected block.
- [ ] Check that it shows the image selector.
- Select an Image block with content.
- [ ] Check that it reads as described before.
- If it doesn't have `alt`, add one (recommended to deactivate VoiceOver since the bottom-sheet is not scree reader friendly yet)
- [ ] Check that VoiceOver reads "Image block. {alt}." on the image block with `alt`.
- Double tap on the block to activate it.
- [ ] Check that the block is now active, and the inner elements are selectable.
